plugins/chat/assets/javascripts/discourse/services/chat-subscriptions-manager.js

```diff
@@ -188,20 +199,66 @@ export default class ChatSubscriptionsManager extends Service {
         if (this.currentUser.ignored_users.includes(busData.username)) {
           channel.currentUserMembership.lastReadMessageId = busData.message_id;
         } else {
-          // Message from other user. Increment trackings state
+          // Message from other user. Increment unread for channel tracking state.
           if (
             busData.message_id >
             (channel.currentUserMembership.lastReadMessageId || 0)
           ) {
             channel.tracking.unreadCount++;
           }
+
+          // Thread should be considered unread if not already.
+          if (busData.thread_id) {
+            if (!channel.threadTrackingOverview.includes(busData.thread_id)) {
+              channel.threadTrackingOverview.push(busData.thread_id);
+            }
+          }
         }
       }
 
       channel.lastMessageSentAt = new Date();
     });
   }
 
+  _onNewThreadMessage(busData) {
+    this.chatChannelsManager.find(busData.channel_id).then((channel) => {
+      channel.threadsManager
+        .find(busData.channel_id, busData.thread_id)
+        .then((thread) => {
+          if (busData.user_id === this.currentUser.id) {
+            // Thread should no longer be considered unread.
+            if (channel.threadTrackingOverview.includes(busData.thread_id)) {
+              channel.threadTrackingOverview =
+                channel.threadTrackingOverview.filter(
+                  (threadId) => threadId !== busData.thread_id
+                );
+            }
+            // TODO (martin) Update lastReadMessageId for thread membership on client.
+          } else {
+            if (this.currentUser.ignored_users.includes(busData.username)) {
+              // TODO (martin) Update lastReadMessageId for thread membership on client.
+            } else {
+              if (
+                this.chat.activeChannel?.activeThread?.id === busData.thread_id
+              ) {
+                // TODO (martin) HACK: We don't yet have the lastReadMessageId on the client,
+                // so if the user is looking at the thread don't do anything to mark it unread.
+              } else {
+                // Message from other user. Thread should be considered unread if not already.
+                if (
+                  !channel.threadTrackingOverview.includes(busData.thread_id)
+                ) {
+                  channel.threadTrackingOverview.push(busData.thread_id);
+                }
```

Comment:
Same, here we could just do:

`channel.threadTrackingOverview.add(bustData.thread_Id)`